python/datafusion/io.py

```diff
@@ -0,0 +1,181 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""IO read functions using global context."""
+
+import pathlib
+
+from datafusion.dataframe import DataFrame
+from datafusion.expr import Expr
+import pyarrow
```

Comment:
Side note: it would be great to use ruff (https://stackoverflow.com/a/77876298) or isort to deterministically and programmatically sort python imports, and validate that in CI. I think isort/ruff would have a newline here between the third-party and first-party imports.

Reply:
there a pre-commit config for ruff linter and formatter
https://github.com/apache/datafusion-python/blob/79c22d6d6c0809e7e93a0a23249baa516dbd8d6f/.pre-commit-config.yaml#L23-L30

Reply:
As the SO answer explains, import sorting isn't currently part of the default `ruff-format` behavior. We'd need to opt-in by adding an `I` element here:
https://github.com/apache/datafusion-python/blob/79c22d6d6c0809e7e93a0a23249baa516dbd8d6f/pyproject.toml#L66